File: books/learn-rust-doing/ch2/cmptext.py
# ファイル名を指定
afile = "../ch1/fizzbuzz_python.txt"
bfile = "../ch1/fizzbuzz_rust.txt"

# ファイルを文字列として読み込む
with open(afile, "r") as fp:
    astr = fp.read()
with open(bfile, "r") as fp:
    bstr = fp.read()

# 念の為余分な空行を削除
astr = astr.strip()
bstr = bstr.strip()

# 比較
if astr == bstr:
    print("OK")
else:
    print("NG")

File: books/learn-rust-doing/ch2/cmptext.rs
// ファイル操作のライブラリーを読む
use std::fs;

fn main() {
    // ファイル名を指定
    let afile = "../ch1/fizzbuzz_python.txt";
    let bfile = "../ch1/fizzbuzz_rust.txt";

    // ファイルを文字列として読む
    let astr = fs::read_to_string(afile).unwrap();
    let bstr = fs::read_to_string(bfile).unwrap();

    // 念の為トリム
    let astr = astr.trim();
    let bstr = bstr.trim();

    // 両者を比較
    if astr == bstr {
        println!("OK");
    } else {
        println!("NG");
    }
}
